mdata/store_cassandra.go

```diff
@@ -357,18 +356,13 @@ func (c *cassandraStore) Search(key string, start, end uint32) ([]iter.Iter, err
 			chunkSizeAtLoad.Value(int64(len(b)))
 			if len(b) < 2 {
 				log.Error(3, errChunkTooSmall.Error())
-				return iters, errChunkTooSmall
+				return itergens, errChunkTooSmall
 			}
 			if chunk.Format(b[0]) != chunk.FormatStandardGoTsz {
 				log.Error(3, errUnknownChunkFormat.Error())
-				return iters, errUnknownChunkFormat
+				return itergens, errUnknownChunkFormat
 			}
-			it, err := tsz.NewIterator(b[1:])
-			if err != nil {
-				log.Error(3, "failed to unpack cassandra payload. %s", err)
-				return iters, err
-			}
-			iters = append(iters, iter.New(it, true))
+			itergens = append(itergens, iter.NewGen(b[1:], uint32(ts)))
```

Comment:
i personally would put all tsz and chunk format things in iter.go but if you feel differently about it, then do it your way it doesn't matter much